third_party/move/move-compiler-v2/tests/checking/typing/loop_result_type.exp

```diff
@@ -81,6 +81,6 @@ module 0x2::M {
     }
     fun t4() {
         let () = loop break;
-        let () = loop if (false) break;
+        let () = while (!false) ();
```

Comment:
Looks like a regression (based on the original source), but is likely ok.

Reply:
It's actually quite interesting, this code diverges, and both are correct in this sense. The new one is slightly better.